### PR TITLE
Move total actions count from index service to basic block store #1234

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1234

### CODE REVIEW

File: blockchain/blockdao.go [hunk 1]
```diff
@@ -120,7 +122,46 @@ func (dao *blockDAO) Start(ctx context.Context) error {
...
+	}
+	return dao.countActions()
+}
+func (dao *blockDAO) countActions() error {
+	totalActions := uint64(0)
+	tipHeight, err := dao.getBlockchainHeight()
+	if err != nil {
+		return err
+	}
+	batch := db.NewBatch()
```
Comment (zjshen14) [inline blockchain/blockdao.go]: Move this to L155

File: blockchain/blockdao.go [hunk 2]
```diff
@@ -120,7 +122,46 @@ func (dao *blockDAO) Start(ctx context.Context) error {
...
+	for i := uint64(1); i <= tipHeight; i++ {
+		hash, err := dao.getBlockHash(i)
+		if err != nil {
+			log.L().Error("Error when get block hash", zap.Error(err))
```
Comment (zjshen14) [inline blockchain/blockdao.go]: Don't need to log if you return the error

File: blockchain/blockdao.go [hunk 3]
```diff
@@ -120,7 +122,46 @@ func (dao *blockDAO) Start(ctx context.Context) error {
...
+			return err
+		}
+		body, err := dao.body(hash)
+		if err != nil {
+			log.L().Error("Error when get block", zap.Error(err))
```
Comment (zjshen14) [inline blockchain/blockdao.go]: ditto

File: blockchain/blockdao.go [hunk 4]
```diff
@@ -120,7 +122,46 @@ func (dao *blockDAO) Start(ctx context.Context) error {
...
+			return err
+		}
+		totalActions += uint64(len(body.Actions))
+		if i%100 == 0 {
```
Comment (zjshen14) [inline blockchain/blockdao.go]: 1000?

File: blockchain/blockdao.go [hunk 5]
```diff
@@ -120,7 +122,46 @@ func (dao *blockDAO) Start(ctx context.Context) error {
...
+			zap.L().Info("loading", zap.Uint64("height", i))
```
Comment (zjshen14) [inline blockchain/blockdao.go]: "loading" -> Counting number of actions

File: blockchain/blockdao.go [hunk 6]
```diff
@@ -120,7 +122,46 @@ func (dao *blockDAO) Start(ctx context.Context) error {
...
+		}
+	}
+	totalActionsBytes := byteutil.Uint64ToBytes(totalActions)
+	batch.Put(blockNS, totalActionsKey, totalActionsBytes, "failed to put total actions")
+	if err := dao.kvstore.Commit(batch); err != nil {
+		log.L().Error(
```
Comment (zjshen14) [inline blockchain/blockdao.go]: ditto

File: blockchain/blockdao.go (hunk 4 above)
Comment (lzxm160) [reply blockchain/blockdao.go]: ok

**Pull request merged**